nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -92,44 +90,54 @@ public CommentManagerImpl(CommentServiceConfig config) {
     }
 
     public List<DocumentModel> getComments(DocumentModel docModel) {
-        Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(
-                ResourceAdapter.CORE_SESSION_CONTEXT_KEY, docModel.getCoreSession());
-        RelationManager relationManager = Framework.getService(RelationManager.class);
-        Graph graph = relationManager.getGraph(config.graphName, docModel.getCoreSession());
-        Resource docResource = relationManager.getResource(config.documentNamespace, docModel, ctxMap);
-        if (docResource == null) {
-            throw new NuxeoException("Could not adapt document model to relation resource ; "
-                    + "check the service relation adapters configuration");
-        }
+        return getComments0(docModel);
```

Comment:
Another elegant solution is to introduce just one method to do the check on _CoreSession_ presence and then call the appropriate `doPrivileged` such as:
```
    protected <S> S doPrivileged(DocumentModel doc, Function<CoreSession, S> function) {
        if (doc.getCoreSession() == null) {
            return CoreInstance.doPrivileged(doc.getRepositoryName(), function);
        }
        return CoreInstance.doPrivileged(doc.getCoreSession(), function);
    }
```
And then just do:
```
public List<DocumentModel> getComments(DocumentModel docModel) {
    return doPrivileged(docModel, s -> {
        // actual method code
    });
}
```
And the same for `getDocumentsForComment`.